rust/agama-server/src/profile/web.rs

```diff
@@ -0,0 +1,270 @@
+    )
+)]
+async fn evaluate(
+    query: Query<ProfileQuery>,
+    profile: String, // jsonnet_or_empty
+) -> Result<String, ProfileServiceError> {
+    let request_has_body = !profile.is_empty() && profile != "null";
+    query.check(request_has_body)?;
+    let profile_string = match query.retrieve_profile()? {
+        Some(retrieved) => retrieved,
+        None => profile,
+    };
+
+    let evaluator = ProfileEvaluator {};
+    let output = evaluator
+        .evaluate_string(&profile_string)
+        .context("Could not evaluate the profile".to_string())?;
+    Ok(output)
+}
+
+#[utoipa::path(
+    post,
+    path = "/autoyast",
+    context_path = "/api/profile",
+    params(ProfileQuery),
+    responses(
+        (status = 200, description = "JSON result of Autoyast profile conversion"),
+        // TODO: "failed to run agama-autoyast" should be a 500 instead, see software/web.rs
+        (status = 400, description = "FIXME some error has happened")
+    )
+)]
+async fn autoyast(
+    query: Query<ProfileQuery>,
+    profile: String, // xml_or_erb_or_empty
+) -> Result<String, ProfileServiceError> {
+    let request_has_body = !profile.is_empty() && profile != "null";
+    if query.url.is_none() || query.path.is_some() || request_has_body {
+        return Err(anyhow::anyhow!(
+            "Only url= is expected, no path= or request body. Seen: url {}, path {}, body {}",
+            query.url.is_some(),
+            query.path.is_some(),
+            request_has_body
+        )
+        .into());
+    }
+
+    let url = Url::parse(query.url.as_ref().unwrap()).map_err(anyhow::Error::new)?;
+    let importer_res = AutoyastProfileImporter::read(&url);
+    match importer_res {
+        Ok(importer) => Ok(importer.content),
+        Err(error) => {
+            // anyhow can be only displayed, not so nice
+            if format!("{}", error).contains("Failed to run") {
+                return Err(ProfileServiceError {
+                    http_status: StatusCode::INTERNAL_SERVER_ERROR,
+                    source: error,
+                });
+            }
+            Err(error.into())
+        }
+    }
+}
+
+#[utoipa::path(
+    post,
+    path = "/execute_script",
+    context_path = "/api/profile",
+    params(ProfileQuery),
+    responses(
+        (status = 200, description = "Script has started"),
+        (status = 400, description = "FIXME some error has happened")
+    )
+)]
+async fn execute_script(
+    query: Query<ProfileQuery>,
+    script: String, // script_or_empty
+) -> Result<Json<()>, ProfileServiceError> {
+    let request_has_body = !script.is_empty() && script != "null";
+    query.check(request_has_body)?;
+    let script_string = match query.retrieve_profile()? {
+        Some(retrieved) => retrieved,
+        None => script,
+    };
+
+    // write to a temporary path and spawn in background and return
+    // May be long running
+    // FIXME: can agama+axum wait for completion and return output?
```

Comment:
Yes, because this action blocks the whole server. You could give [tokio::process::Command](https://docs.rs/tokio/latest/tokio/process/struct.Command.html) a try. Alternatively, if it does not work, we could use a separate Tokio task.